lib/spack/spack/store.py

```diff
@@ -52,17 +119,21 @@ class Store(object):
 
     Args:
         root (str): path to the root of the install tree
+        sbang_root (str): path to the root of the install tree without padding;
+            the sbang script has to be installed here to work with padded roots
```

Comment:
Can we try not to couple the concepts of `sbang` and the `store`?  I think this is too tight.  The concept of padding has to do with the store, but `sbang` just happens to be a script.  Maybe call this `unpadded_root`?  I don't *really* like that because I would think the root is the topmost thing.

Does this put the DB in the padded or unpadded root?  IMO the DB and all top-level stuff should be at the top rather than buried under the padding.  Do we need to put the index, locks, etc. way down in the padded directory?  Seems more intuitive if `.spack-db` is at the top and the padding is just a directory that happens to be the prefix of all installed packages.

Reply:
As we discussed, we need to keep `.spack-db` in the padded prefix.

I've updated the name to `unpadded_root`.